### PR TITLE
Allow `dataclasses.dataclass` to be used directly on extension types

### PR DESCRIPTION
**Pull request by da-woods**
Also implements the same for `functools.total_ordering`.

Closes https://github.com/cython/cython/issues/5292 - as discussed in that issue, if the directives are taken from the `cython` module, `cclass` is automatically implied.

### CODE REVIEW

**Comment by da-woods** [comment]
@scoder this might be a good candidate for the next beta (given the issue is marked as 3.0). Sorry that it's slightly late in appearing

**Comment by h-vetinari** [comment]
From the sidelines, at some point the feature work for 3.0 should wind down (resp. that work should target 3.1), with only regression fixes going into 3.0 anymore, until that's out.

**Comment by da-woods** [comment]
> From the sidelines, at some point the feature work for 3.0 should wind down (resp. that work should target 3.1), with only regression fixes going into 3.0 anymore, until that's out.

Yeah I agree. This is the last feature I'll try to add to 3.0 (can't speak for anyone else of course...). In my defence, it's sort of half a feature - a usability fix-up for something we've already added.

File: Cython/Compiler/Options.py [hunk 1]
```diff
@@ -391,7 +391,7 @@ class DEFER_ANALYSIS_OF_ARGUMENTS:
     'fast_gil': ('module',),
     'iterable_coroutine': ('module', 'function'),
     'trashcan' : ('cclass',),
-    'total_ordering': ('cclass', ),
+    'total_ordering': ('cclass', 'class'),
     'dataclasses.dataclass' : ('class', 'cclass',),
```
Comment (scoder) [inline Cython/Compiler/Options.py:395]: Let's keep the order consistent here, given that both are even right next to each other.
```suggestion
    'total_ordering': ('class', 'cclass'),
    'dataclasses.dataclass' : ('class', 'cclass'),
```

**Review by scoder — COMMENTED** [review]
Does this also work with `import functools as different_name`?
(If not, I don't think it necessarily has to work in the first iteration – we can fix that later.)

**Comment by scoder** [comment]
> at some point the feature work for 3.0 should wind down

I agree, but this is not a new feature. It was found to be a desirable improvement of an existing 3.0 feature before.

**Comment by scoder** [comment]
Nice, thanks!

**Comment by da-woods** [comment]
> Does this also work with `import functools as different_name`?

I'm fairly sure this should work, but it isn't tested

**Pull request merged**